src/item_group.cpp

```diff
@@ -418,7 +418,9 @@ void Item_modifier::modify( item &new_item, const std::string &context ) const
     if( container != nullptr ) {
         cont = container->create_single( new_item.birthday() );
     }
-    if( cont.is_null() && new_item.type->default_container.has_value() ) {
+    if( container != nullptr && cont.is_null() ) {
+        cont = container->create_single( new_item.birthday() );
+    } else if( cont.is_null() && new_item.type->default_container.has_value() ) {
```

Comment:
I think you should simplify this as discussed on discord. Apart from the code duplication and possible duplicated execution, it's also harder to understand what exactly is supposed to happen here.